init.lua

```diff
@@ -1220,6 +1220,97 @@ function Form:update_where(func)
     end
 end
 
+local embed_create_ctx_mt = {}
+
```

Comment:
All of these changes could be moved into a different file, since it's getting long. Perahps that's best for a different PR/Issue

Reply:
Just moved it! (several review comments are now marked as outdated)

Reply:
Thanks